packages/wonder-blocks-core/components/unique-id-provider.test.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import * as React from "react";
+import * as ReactDOMServer from "react-dom/server";
+
+import {mount} from "enzyme";
+
+import View from "./view.js";
+
+import SsrIDFactory from "../util/ssr-id-factory.js";
+import UniqueIDFactory from "../util/unique-id-factory.js";
+import UniqueIDProvider from "./unique-id-provider.js";
+
+describe("UniqueIDProvider", () => {
+    describe("mockOnFirstRender is default (false)", () => {
+        test("initial render is nothing on server", () => {
+            // Arrange
+            const children = jest.fn(() => null);
+            const nodes = <UniqueIDProvider>{children}</UniqueIDProvider>;
+
+            // Act
+            ReactDOMServer.renderToStaticMarkup(nodes);
+
+            // Assert
+            expect(children).not.toHaveBeenCalled();
+        });
+
+        test("initial render is nothing on client", () => {
+            // Arrange
+            const children = jest.fn(() => null);
+            const nodes = <UniqueIDProvider>{children}</UniqueIDProvider>;
+
+            // Act
+            mount(nodes);
+
+            // Assert
+            expect(children).toHaveBeenCalledTimes(1);
+            expect(children).not.toHaveBeenCalledWith(SsrIDFactory);
```

Comment:
Would
```
expect(children.mock.calls[0][0]).toBeInstanceOf(UniqueIDFactory);
```
be try in this test?

Reply:
I've added some additional commentary and reworked the asserts for this.